### DISCUSSION TITLE
Tag-Affixes for Monorepo Dependencies

### DISCUSSION THREAD

**Discussion by SebastianGoeb** [renovatebot/renovate#17405] [Request Help] [answered]
Can Renovate handle tag-based dependencies (e.g. gitlabci-include) for monorepos? It's pretty common to prefix/suffix tags, e.g.
```
app1/1.2.3
app1/1.3.0
app2/6.2.0
app2/6.2.1
```
so that multiple version histories can co-exist in one repo (app2 doesn't get a new version when app1 is released and vice versa). That's not semver compatible though, and I can't find any config options in the docs that might allow me to strip a tag-prefix/-suffix, to configure a tag regex match, or something similar.

Whenever possible, pushing the build artefacts of a monorepo to a registry (nexus, dockerregistry, chartmuseum, npm, what-have-you) is preferable, because then the artefact id/name will disentangle the versions and separate them into clean isolated series. However, sometimes that's not possible, specifically whenever the dependency is the original tag itself (gitlabci-include, etc.). For example a shared CI/CD pipeline project.

Has this been considered as a use case? Does this already exist, and I'm just blind? Would this be an interesting feature?

**Comment by rarkins** [comment, marked as answer]
Check whether Renovate's `extractVersion` feature suits your needs to extract the semver part from the longer tag

**Reply by SebastianGoeb** [reply]
Ah yes, that's what I was looking for. Thanks for the pointer.
